reactor-core/src/main/java/reactor/core/publisher/EmitterProcessor.java

```diff
@@ -310,7 +310,7 @@ public Emission tryEmitNext(T t) {
 		}
 
 		if (!q.offer(t)) {
-			return Emission.FAIL_OVERFLOW;
+			return subscribers == EMPTY ? Emission.FAIL_ZERO_SUBSCRIBER : Emission.FAIL_OVERFLOW;
```

Comment:
shouldn't we check `subscribers == EMPTY` even before attempting to lazily create a queue? Otherwise, it still returns `OK` if the queue has the capacity but there are no subscribers

Reply:
One of the advertised features of the `EmitterProcessor` is the capacity to buffer a (few?) elements into that queue before failing with overflow, including when trying to emit before anybody's listening. If the queue has the capacity but there are no listeners, the first listener will get the early emission from the queue, so this is intended.